pytorch_lightning/plugins/precision/native_amp.py

```diff
@@ -96,7 +96,8 @@ def pre_optimizer_step(
                 " To request, please file a Github issue in PyTorch and tag @mcarilli"
             )
         result = lambda_closure()  # native amp does not support closures
-        self.scaler.unscale_(optimizer)
+        if not model.automatic_optimization:
```

Comment:
Why is this if added? `step` below will not unscale again if it has been done manually already

Reply:
@carmocca I don't like it either. This makes the hacks that exist here even more hacky. 
The PR aims at solving the bug that unscaling is happening after gradient clipping. So to fix this, one can move the unscaling to an earlier point, which we choose to do directly after backward (post_backward). However, we ONLY want to do that in the case of automatic optimization, because gradient clipping does not exist for manual backward and we need to keep the unscaling in the optimizer_step here because the user of manual optimization will call the LightningOptimizer.step themselves. 

Bottom line: we have unscale in two locations, but only one is ever active depending on whether we are in automatic or manual optimization.